pkg/utils/utils.go

```diff
@@ -157,3 +159,9 @@ func ResolveNodePoolFromNodeClaim(ctx context.Context, kubeClient client.Client,
 	// There will be no nodePool referenced inside the nodeClaim in case of standalone nodeClaims
 	return nil, nil
 }
+
+func CamelToSnakeCase(str string) string {
```

Comment:
Any reason to not just make the message kind itself return its value in snake case rather than having to do this conversion in-code only for this metric